tests/test_workers/test_pkg_managers/test_general.py

```diff
@@ -149,3 +158,20 @@ def test_update_request_env_vars_failed(mock_patch, side_effect, expected_error)
     mock_patch.side_effect = [side_effect]
     with pytest.raises(CachitoError, match=expected_error):
         update_request_env_vars(1, {"environment_variables": {}})
+
+
+@mock.patch("cachito.workers.pkg_managers.general.nexus.upload_raw_component")
+@pytest.mark.parametrize("is_request_repo", [True, False])
+def test_upload_raw_package(mock_upload, caplog, is_request_repo):
+    """Check Nexus upload calls."""
+    name = "name"
+    path = "fakepath"
+    dest_dir = "name/varsion"
+    filename = "name.tar.gz"
+
+    upload_raw_package(name, path, dest_dir, filename, is_request_repo)
```

Comment:
> Do you have any idea why are tests failing after the second refactor? They stopped working with `Refactor _download_raw_component` commit and when I'm testing it manually in my local development environment (I mean not running unit tests, but changing code a bit and creating regular curl request to Cachito), the new changes seem to work, so I'm wondering that maybe I'm patching something it those tests poorly.

Not sure, `mock.patch` tends to be rather fragile :disappointed: 

The only difference I can see is that the previous test used `pip.upload_raw_package()` while this one uses the imported function directly, but I'm not sure if that makes any difference

Reply:
Turns out, `download_binary_file` has to be patched twice, once called from `general` and once from `pip`.